src/linear_algebra/affine_space.lean

```diff
@@ -212,6 +233,77 @@ end
 
 end finset
 
+section affine_independent
+
+open add_torsor
+
+variables (k : Type*) (V : Type*) {P : Type*} [ring k] [add_comm_group V] [module k V]
+variables [affine_space k V P] {ι : Type*}
+
+/-- An indexed family is said to be affinely independent if no
+nontrivial weighted subtractions (where the sum of weights is 0) are
+0. -/
+def affine_independent (p : ι → P) : Prop :=
+∀ (s : finset ι) (w : ι → k) (h : ∑ i in s, w i = 0), s.weighted_vsub V p w = 0 → ∀ i ∈ s, w i = 0
+
+/-- A family with at most one point is affinely independent. -/
+lemma affine_independent_of_subsingleton [subsingleton ι] (p : ι → P) :
+  affine_independent k V p :=
+λ s w h hs i hi, fintype.eq_of_subsingleton_of_sum_eq h i hi
+
+/-- A family is affinely independent if and only if the differences
+from a base point in that family are linearly independent. -/
+lemma affine_independent_iff_linear_independent_vsub (p : ι → P) (i1 : ι) :
+  affine_independent k V p ↔ linear_independent k (λ i : {x // x ≠ i1}, (p i -ᵥ p i1 : V)) :=
+begin
+  split,
+  { intro h,
+    rw linear_independent_iff',
+    intros s g hg i hi,
+    set f : ι → k := λ x, if hx : x = i1 then -∑ y in s, g y else g ⟨x, hx⟩ with hfdef,
```

Comment:
I think you can define `f` in terms of `function.update`. That might simplify some properties.

Reply:
Oh wait, `g` depends on the proof `hx`. Disregard this.